src/main/java/org/broadinstitute/hellbender/tools/pon/coverage/pca/HDF5PCACoveragePoNCreationUtils.java

```diff
@@ -531,178 +430,173 @@ static ReadCountCollection removeTargetsWithTooManyZeros(final ReadCountCollecti
         }
     }
 
+    private static long countZeroes(final double[] data) {
+        return DoubleStream.of(data).filter(d -> d == 0.0).count();
+    }
+
     /**
-     * Remove columns that have too many counts equal to 0.
-     * <p>
-     *     It will return a copy of the input read-count collection with such columns dropped.
-     * </p>
+     * Creates a new read-count collection that is a copy of the input but dropping columns with extreme median coverage.
      *
-     * @param readCounts the input read counts.
-     * @param maximumColumnZeros maximum number of counts equal to 0. per column tolerated.
-     * @return never {@code null}. It might be a reference to the input read-counts if there is
-     *   is no column to be dropped.
+     * @param readCounts the input read-counts.
+     * @param extremeColumnMedianCountPercentileThreshold bottom percentile to use as an exclusion threshold.
+     * @return never {@code null}. It might be a reference to the input read-counts if
+     * there are no columns to be dropped.
      */
     @VisibleForTesting
-    static ReadCountCollection removeColumnsWithTooManyZeros(final ReadCountCollection readCounts, final int maximumColumnZeros, final Logger logger) {
-
+    static ReadCountCollection removeColumnsWithExtremeMedianCounts(final ReadCountCollection readCounts, final double extremeColumnMedianCountPercentileThreshold, final Logger logger) {
+        final List<String> columnNames = readCounts.columnNames();
         final RealMatrix counts = readCounts.counts();
+        final double[] columnMedians = MatrixSummaryUtils.getColumnMedians(counts);
 
-        final Set<String> columnsToKeep = IntStream.range(0, counts.getColumnDimension()).boxed()
-                .filter(i -> countZeroes(counts.getColumn(i)) <= maximumColumnZeros)
-                .map(i -> readCounts.columnNames().get(i)).collect(Collectors.toSet());
+        // Calculate thresholds:
+        final double bottomExtremeThreshold = new Percentile(extremeColumnMedianCountPercentileThreshold).evaluate(columnMedians);
+        final double topExtremeThreshold = new Percentile(100 - extremeColumnMedianCountPercentileThreshold).evaluate(columnMedians);
 
-        final int columnsToDropCount = readCounts.columnNames().size() - columnsToKeep.size();
-        if (columnsToDropCount == 0) {
-            logger.info(
-                    String.format("No count column dropped due to zero counts; there is no column with a large number of targets with zero counts (<= %d of %d) ", maximumColumnZeros, readCounts.targets().size()));
+        // Determine kept and dropped column sets.
+        final Set<String> columnsToKeep = new HashSet<>(readCounts.columnNames().size());
+        final Map<String, Double> columnsToDrop = new LinkedHashMap<>((int) (extremeColumnMedianCountPercentileThreshold * 4.0 / 100.0));
```

Comment:
Where does the `percentileThreshold * 4/100` capacity come from?  I would have thought it was `2 * (percentileThreshold/100) * readCounts.columnNames().size()`.